src/pages/Election.tsx

```diff
@@ -58,11 +67,20 @@ const ManagementTools: FC<ElectionSubpage> = ({
           appearance="primary"
           icon={<Icon icon="trash" />}
           color="red"
+          onClick={() => {
+            setIsOpen(true);
+          }}
         >
           Delete Election
         </IconButton>
       </ButtonToolbar>
-      {/* TODO: Add delete election modal. */}
+      <ConfirmModal
+        modalTitle="Delete Election"
+        modalBody="Do you want to delete this election?"
+        callBackFunc={() => deleteElection(id)}
```

Comment:
The history call we discussed can be placed in a `then()` statement after deleteElection resolves, like `deleteElection.then(() => { history.push('/setup') }`